src/permission-set.js

```diff
@@ -16,523 +16,809 @@
+     * @property authorizations
+     * @type {Object}
+     */
+    this.authorizations = {}
+    /**
+     * The URL of the corresponding ACL resource, at which these permissions will
+     * be saved.
+     * @property aclUrl
+     * @type {String}
+     */
+    this.aclUrl = aclUrl
+    /**
+     * Optional request host (used by checkOrigin())
+     * @property host
+     * @type {String}
+     */
+    this.host = options.host
+    /**
+     * Initialize the agents / groups / resources indexes.
+     * @property index
+     * @type {Object}
+     */
+    this.index = {
+      'agents': {},
+      'groups': {}  // Also includes Public permissions
+    }
+    /**
+     * RDF Library (optionally injected)
+     * @property rdf
+     * @type {RDF}
+     */
+    this.rdf = options.rdf
+    /**
+     * Whether this permission set is for a 'container' or a 'resource'.
+     * Determines whether or not the inherit/'acl:default' attribute is set on
+     * all its Authorizations.
+     * @property resourceType
+     * @type {String}
+     */
+    this.resourceType = isContainer ? CONTAINER : RESOURCE
+    /**
+     * The URL of the resource for which these permissions apply.
+     * @property resourceUrl
+     * @type {String}
+     */
+    this.resourceUrl = resourceUrl
+    /**
+     * Should this permission set enforce "strict origin" policy?
+     * (If true, uses `options.origin` parameter)
+     * @property strictOrigin
+     * @type {Boolean}
+     */
+    this.strictOrigin = options.strictOrigin
+    /**
+     * Contents of the request's `Origin:` header.
+     * (used only if `strictOrigin` parameter is set to true)
+     * @property origin
+     * @type {String}
+     */
+    this.origin = options.origin
+    /**
+     * Solid REST client (optionally injected), used by save() and clear().
+     * @type {SolidWebClient}
+     */
+    this.webClient = options.webClient
+
+    // Init the functions for deriving an ACL url for a given resource
+    this.aclUrlFor = options.aclUrlFor
+      ? options.aclUrlFor
+      : PermissionSet.defaultAclUrlFor
+    this.aclUrlFor.bind(this)
+    this.isAcl = options.isAcl ? options.isAcl : PermissionSet.defaultIsAcl
+    this.isAcl.bind(this)
+
+    // Optionally initialize from a given parsed graph
+    if (options.graph) {
+      this.initFromGraph(options.graph)
+    }
+  }
+
   /**
-   * Hashmap of all Authorizations in this permission set, keyed by a hashed
-   * combination of an agent's/group's webId and the resourceUrl.
-   * @property authorizations
-   * @type {Object}
+   * Returns the corresponding ACL uri, for a given resource.
+   * This is the default `aclUrlFor()` method that's used by
+   * PermissionSet instances, unless it's overridden in options.
+   * @method defaultAclUrlFor
+   * @param resourceUri {String}
+   * @return {String} ACL uri
    */
-  this.authorizations = {}
+  static defaultAclUrlFor (resourceUri) {
+    if (this.isAcl(resourceUri)) {
```

Comment:
I think you've got the semantics of static methods mixed up. For static methods, `this` is bound to the class `Foo`. For instance methods, it's bound to the current instance of `Foo`. So `this.isAcl` will be undefined. You should probably just remove the `static` qualifier since this is really an instance method.